src/rust/engine/fs/store/src/lib.rs

```diff
@@ -236,6 +236,20 @@ impl RemoteStore {
       .map(|&()| ())
   }
 
+  async fn remote_writer(
+    remote_store: &remote::ByteStore,
+    digest: Digest,
+    file: tokio::fs::File,
+  ) -> Result<tokio::fs::File, String> {
```

Comment:
The fact that this returns a `String` ErrorType is the issue. I do that because `write_using` already returns `String` errors.

Not entirely sure the best way to handle propogating custom error type through `write_using` :thinking: